server/src/main/java/org/opensearch/monitor/os/OsProbe.java

```diff
@@ -588,11 +598,18 @@ public static OsProbe getInstance() {
         return OsProbeHolder.INSTANCE;
     }
 
-    OsProbe() {
+    private final Logger logger;
+
+    private boolean isCpuStatWarnignsLogged = false;
```

Comment:
@andrross @reta Also am I overthinking it here?

Reply:
I think a bit. `Oservice` creates this class ones and `OsService` creates in `MontorSevice` which is created by `NodeService` during boot up of the node.

Reply:
In my opinion the thread safe version using AtomicBoolean isn't really any more complicated or less performant than using a primitive boolean so there isn't really a trade off and we might as well do it :)

Reply:
It *is* 1 less line of code.

Reply:
my 5 cents, `AtomicBoolean` is simple to use (at least with respect to CAS) but it does class to `Unsafe`, I think we try to make sure we are not flooding logs with same warning, using simple boolean in thread safe manner needs `volatile` but it is not a big of deal if the warnings will be printed out a few times. I would keep simple :-)

Reply:
Do we do boolean the way it's done in this PR elsewhere?

If I must decide I'd like `AtomicBoolean`, sorry to make this a bigger problem than it is @willyborankin, do tell me if I am annoying.

Reply:
> but it does calls to `Unsafe`

@reta Just curious, what is the implication of this? I know this case mostly comes down to style preference (I'm with @dblock...AtomicBoolean is one less line of code!), but are there situations where the usage of Unsafe inside an Atomic* variant would be problematic?

Reply:
@andrross ahh, not, it is not problematic per se, more like "why?" if simpler constructs fit. Let's go with `AtomicBoolean` (to make @dblock happy :D) - it will do the job just fine

Reply:
@dblock you are not annoying, I understand your concerns. I added `AtomicBoolean` and check using `getAndSet(true) == false`. IMHO I'm sure it won't affect a booting node start time. But If it is just ping me directly I will fix it by switching to primitive `boolean` or `volatile`.

Reply:
I prefer `AtomicBoolean`. It reminds me of my childhood living in the USSR near some very dangerous power plants. Thanks for hanging out with me here!

Reply:
> I prefer `AtomicBoolean`. It reminds me of my childhood living in the USSR near some very dangerous power plants. Thanks for hanging out with me here!

I feel like we've all learned a lot here in multiple dimensions ;)